src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,245 @@
+from tqec.circuit.schedule import ScheduledCircuit
+from tqec.circuit.qubit_map import QubitMap
+from tqec.plaquette.plaquette import Plaquette
+from tqec.plaquette.qubit import PlaquetteQubits
+from tqec.plaquette.qubit import SquarePlaquetteQubits
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from stim import Circuit as stim_Circuit
+
+
+class BasisEnum(Enum):
+    X = 'x'
+    Y = 'y'
+    Z = 'z'
+
+
+class ExtendedBasisEnum(Enum):
+    X = BasisEnum.X.value
+    Y = BasisEnum.Y.value
+    Z = BasisEnum.Z.value
+    H = 'h'
+
+
+@dataclass
+class RPNG:
+    """Organize a single RPNG value
+
+    ----
+    Simplified RPNG format:
+
+    -z1-
+    rpng
+    
+    (r) data qubit reset basis or h or -  
+    (p) data basis for the controlled operation (x means CNOT controlled on the ancilla and targeting the data qubit, y means CY, z means CZ)  
+    (n) time step (positive integers, all distinct, typically in 1-5)  
+    (g) data qubit measure basis or h or -
+
+    Assumptions on the circuit:
+    - if not otherwise stated, a basis can be {x,y,z}
+    - the ancilla is always the control qubit for the CNOT, CY, or CZ gates
+    - time step of r same as ancilla reset (always 0)
+    - time step of g same as ancilla measurement (by default 6)
+
+    ----
+    Extended RPNG format:
+
+    The extra field (a) is used together with (p) to determine the 2Q gate involving the
+    ancilla and data qubits. For example:
+    - a=z, p=x --> CNOT controlled by the ancilla and targeting the data
+    - a=z, p=z --> CZ   controlled by the ancilla and targeting the data
+    - a=y, p=z --> CY   controlled by the data    and targeting the ancilla
+    By default a=z.
```

Comment:
This is not needed anymore I think.
```suggestion
```

Reply:
The extra field (a) does not affect the RPNG description when left to its default value z.
We can move all generalizations of the basic RPNG description somewhere else, but I thought that they are harmless and may be helpful to users in need to customize the plaquette circuits.

Reply:
The extra field:

- is 4 more characters to type each time a user wants a Plaquette,
- makes the code more complex.

We can keep it, but if we do I would advocate for it to be completely separate from the "normal" `RPNG` notation. Basically, have `RPNG` only represent a 4-character string, and have another class (`ERPNG`? `RPNGA`? `RAPNG`?) represent the 5-character notation. The 4-character version will be used extensively in the code, and so it should be easy to read, the simplest possible, and as robust as possible, which is in my opinion incompatible with having both the 4- and 5-character approaches in the same implementation.

Reply:
5-character description moved to rapng.py